hbase-rest/src/main/java/org/apache/hadoop/hbase/rest/MultiRowResource.java

```diff
@@ -65,15 +73,33 @@ public MultiRowResource(TableResource tableResource, String versions, String col
   @GET
   @Produces({ MIMETYPE_XML, MIMETYPE_JSON, MIMETYPE_PROTOBUF, MIMETYPE_PROTOBUF_IETF })
   public Response get(final @Context UriInfo uriInfo,
-    final @HeaderParam("Encoding") String keyEncodingHeader) {
+    final @HeaderParam("Encoding") String keyEncodingHeader,
+    @QueryParam(Constants.SCAN_FILTER_B64) @Encoded String paramFilterB64,
```

Comment:
POST method would have been better for sending the filter as JSON, but that would require a new API to avoid causing incompatibility with the current implementation. So let's continue with it base64

Reply:
I have considered that, but you must not use a POST operation to read data in a REST interface.
The HTTP operations must mutch the semantics of the call, otherwise it's not a REST interface, just some kind of HTTP RPC interface.